solr/solrj/src/java/org/apache/solr/client/solrj/request/beans/PrepareCoreRecoveryPayload.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.client.solrj.request.beans;
+
+import org.apache.solr.common.annotation.JsonProperty;
+import org.apache.solr.common.util.ReflectMapWriter;
+
+public class PrepareCoreRecoveryPayload implements ReflectMapWriter {
+    @JsonProperty
+    public String nodeName;
```

Comment:
Should this be marked required?   Isn't that a property mapping option?

Reply:
I suspect so, but I'm not certain enough to mark it that way here.

The apispec file doesn't explicitly say it's required:

```
    "prep-recovery": {
      "type": "object",
      "additionalProperties": true,
      "description": "An internal API used by the Collections API. This should not be used directly by end-users."
    },
```

And there's no null-checking or other sanity check in PrepRecoveryOp where this value is actually used that'd give me any confidence.

Anyway, to avoid being overly restrictive, I left it as it was documented in the apispec for now.